content/en/docs/tasks/debug/debug-cluster/troubleshoot-kubectl.md

```diff
@@ -0,0 +1,144 @@
+
+```
+
+If you see `Unable to connect to the server: dial tcp <server-ip>:8443: i/o timeout`,
+instead of `Server Version`, you need to troubleshoot kubectl connectivity with your cluster.
+
+If you have installed the `kubectl` but your shell gives the `command not found` error
+make sure that the `$PATH` environment variable is correctly configured and you have
+installed the kubectl by following the [official documentation for installing kubectl](/docs/tasks/tools/#kubectl).
+
+## Check VPN connectivity
+
+If you are using a Virtual Private Network (VPN) to access your Kubernetes cluster,
+make sure that your VPN connection is active and stable. Sometimes, VPN disconnections
+can lead to connection issues with the cluster. Reconnect to the VPN and try accessing
+the cluster again.
+
+## Check kubeconfig
+
+The `kubectl` requires a `kubeconfig` file to connect to a Kubernetes cluster. The
+`kubeconfig` file is usually located under the `~/.kube/config` directory. Make sure
+that you have a valid `kubeconfig` file. If you don't have a `kubeconfig` file, you can
+create and configure it via `kubectl config` command. If you have deployed your Kubernetes
+cluster on a cloud platform and lost your `kubeconfig` file, you can re-generate it using
+your cloud provider's tools. Refer the cloud provider's documentation for re-generating
+the `kubeconfig` file.
+
+Check if the `$KUBECONFIG` environment variable is configured correctly. You can set
+`$KUBECONFIG`environment variable or use the `--kubeconfig` parameter with the kubectl
+to specify the directory of a `kubeconfig` file.
+
+## Authentication and authorization
+
+If the kubectl is returning an error regarding the authentication token or authentication
+server, validate the `$KUBERNETES_SERVER` and`$KUBERNETES_TOKEN` environment variables.
+
+Make sure that you are using the valid user credentials. And you have the permission to
+access the resource that you have requested.
+
+## Verify contexts
+
+Kubernetes supports [multiple clusters and contexts](/docs/tasks/access-application-cluster/configure-access-multiple-clusters/).
+Ensure that you are using the correct context to interact with your cluster.
+
+List available contexts:
+
+```shell
+kubectl config get-contexts
+```
+
+Switch to the appropriate context:
+
+```shell
+kubectl config use-context <context-name>
+```
+
+## API server and load balancer
+
+The {{<glossary_tooltip text="kube-apiserver" term_id="kube-apiserver">}} server is the
+central component of a Kubernetes cluster. If the API server or the load balancer it uses
+is not reachable or not responding, you won't be able to interact with the cluster.
+
+Check the if the API server's host is reachable by using `ping` command. Check cluster's
+network connectivity and firewall. Or if your are using a cloud provider for deploying
```

Comment:
The previous two checks are about network connectivity issues, and the third one is about API server status. I'd not recommend using "Or" here...